File: presentation/07_JSON/json.m
// Topic: JSON Handling
// Parse JSON string

#import <Foundation/Foundation.h>

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        NSString *jsonStr = @"{\"name\": \"Ardium\", \"vers\": 2}";
        NSData *jsonData = [jsonStr dataUsingEncoding:NSUTF8StringEncoding];
        
        NSDictionary *json = [NSJSONSerialization JSONObjectWithData:jsonData options:0 error:nil];
        
        NSLog(@"Name: %@", json[@"name"]);
        NSLog(@"Version: %@", json[@"vers"]);
    }
    return 0;
}

File: presentation/07_JSON/json.go
// Topic: JSON Handling
// Parse JSON string

package main

import (
	"encoding/json"
	"fmt"
)

type Config struct {
	Name string `json:"name"`
	Vers int    `json:"vers"`
}

func main() {
	jsonStr := `{"name": "Ardium", "vers": 2}`
	var config Config

	json.Unmarshal([]byte(jsonStr), &config)

	fmt.Printf("Name: %s\n", config.Name)
	fmt.Printf("Version: %d\n", config.Vers)
}
